### PR TITLE
jackliu-gear-academy

### PR DESCRIPTION
**Pull request by Jackliu-miaozi**

### CODE REVIEW

**Comment by Jackliu-miaozi** [comment]
good

**Comment by shamilsan** [comment]
Hi @Jackliu-miaozi 

You are to create a PR to be merged into your fork, not the upstream one. See the homework rules: https://academy.gear.rs/getting-started/homework-rules

> Submit each homework as a separate pull request (PR) from a feature branch derived from the master or another working branch. Note that the target branch for the PR should be master in your repository, not in the template one.

File: contracts/01-tamagotchi/io/src/lib.rs [hunk 1]
```diff
@@ -10,29 +10,45 @@ use scale_info::TypeInfo;
...
 pub enum TmgAction {
     // TODO: 2️⃣ Add `Name` and `Age` actions that set the name and age
+    Name,
+    Age,
 }
 
+#[derive(Encode, Decode, TypeInfo)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
+pub struct  TmgInit {
```
Comment (shamilsan) [inline contracts/01-tamagotchi/io/src/lib.rs:29]: This structure seems to not be used.

File: contracts/01-tamagotchi/src/lib.rs [hunk 2]
```diff
@@ -1,19 +1,46 @@
 #![no_std]
 
 #[allow(unused_imports)]
-use gstd::prelude::*;
+use gstd::{prelude::*,msg,exec};
```
Comment (shamilsan) [inline contracts/01-tamagotchi/src/lib.rs:4]: Prefer using `rustfmt` before committing.
```suggestion
use gstd::{exec, msg, prelude::*};
```

**Review by shamilsan — COMMENTED** [review]
Also, need to cover the actions processing by tests.

File: contracts/01-tamagotchi/io/src/lib.rs (hunk 1 above)
Comment (Jackliu-miaozi) [reply contracts/01-tamagotchi/io/src/lib.rs:29]: Thank you.  I learning GitHub and Gear protocol. Thank you for your help.

File: contracts/01-tamagotchi/src/lib.rs (hunk 2 above)
Comment (Jackliu-miaozi) [reply contracts/01-tamagotchi/src/lib.rs:4]: Thank you.

**Pull request closed without merging**